python/tk_desktop/desktop_window.py

```diff
@@ -1353,10 +1347,60 @@ def _sync_success(self, config_path, descriptor):
         except Exception as e:
             log.exception("Unexpected error while launching Python:")
             self._launch_failed(str(e))
+        else:
+            # and remember what we launched for next time
+            self._save_setting("project_id", self.current_project["id"], site_specific=True)
+            # Banners might need to be updated, we might have picked a configuration that has been
+            # updated.
+            self._update_banners()
         finally:
             if "SHOTGUN_DESKTOP_CURRENT_USER" in os.environ:
                 del os.environ["SHOTGUN_DESKTOP_CURRENT_USER"]
 
+    def _launch_failed(self, message):
+        message = ("%s"
+                   "\n\nTo resolve this, open Shotgun in your browser\n"
+                   "and check the paths for this Pipeline Configuration."
+                   "\n\nFor more details, see the console." % message)
+        self.project_overlay.show_error_message(message)
+
+    def bootstrap_progress_callback(self, value, msg):
+        self.project_overlay.report_progress(value, msg)
+
+    def _pick_pipeline_configuration(self, pipeline_configurations, requested_pipeline_configuration_id, project):
+        """
```

Comment:
This isn't new code. It used to be define between two methods, but now the above method and below method have merged together so this got displaced at the bottom now.